shell/client/grainview.js

```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```

Comment:
I think we should probably keep it as just a string that the app provides. Attempting to parse the query could break some apps that do stupid things like use the same query parameter name multiple times, and I don't think it gains us anything.

We my want to consider what happens if the app sends "../foo" as the path... but not now.